src/providers/OrganizationProvider.tsx

```diff
@@ -116,15 +117,20 @@ export default function OrganizationProvider({ children }: OrganizationProviderP
       }
       if (orgToUse) {
         setSelectedOrganization(orgToUse);
-        PreferencesService.updateUserPreferences({ lastVisitedOrg: orgToUse.id });
       }
       if (organizationId) {
         query.delete('organizationId');
         // preserve other url params
         history.push(getLocation(location.pathname, location, query.toString()));
       }
     }
-  }, [organizations, selectedOrganization, query, location, history, userPreferences]);
+  }, [organizations, selectedOrganization, query, location, history, userPreferences, userBootstrapped]);
+
+  useEffect(() => {
+    if (selectedOrganization?.id && userPreferences.lastVisitedOrg !== selectedOrganization.id) {
+      updateUserPreferences({ lastVisitedOrg: selectedOrganization.id });
+    }
+  }, [selectedOrganization?.id, updateUserPreferences, userPreferences.lastVisitedOrg]);
```

Comment:
separate useEffect to keep track of last selected org in user preference. this is cleaner and avoids redundant updates